text/0000-c-string-construction.md

```diff
@@ -0,0 +1,72 @@
+- Feature Name: c_string_from_iter
+- Start Date: 2015-02-26
+- RFC PR:
+- Rust Issue:
+
+# Summary
+
+Amend the methods available to construct a `CString` to improve composability
+and follow the conventions emerging elsewhere in the standard library.
+
+# Motivation
+
+The implementation of RFCs [#592][rfc 592] and [#840][rfc 840] has resolved
+most of the issues with the design of `std::ffi::CString`, but some new APIs
+have been created at the same time, needing stabilization. This proposal
+aims at addressing the following issues:
+
+1. The `IntoBytes` trait does not seem wholly justified: it falls short of
+   supporting `IntoIterator`, and has become yet another special-interest trait
+   to care about when designing APIs producing string-like data. 
```

Comment:
Note that this trait was introduced but largely only as a temporary measure. With [generic conversion traits](https://github.com/rust-lang/rfcs/pull/529) this would be much less ad-hoc. The semi-current-plan is to leave `IntoBytes` as `#[unstable]` until we have conversion traits at which point we can switch to using those.